### PR TITLE
Issue1252: vault write broken

### PR DESCRIPTION
**Pull request by eikenb**
Writing to a Vault KV v1 and v2 were both broken.

### CODE REVIEW

File: dependency/vault_write_test.go [hunk 1]
```diff
@@ -71,6 +71,69 @@ func TestNewVaultWriteQuery(t *testing.T) {
...
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		_, _, err = wq.Fetch(clients, &QueryOptions{})
+		if err != nil {
+			fmt.Println(err)
+		}
+
+		rq, err := NewVaultReadQuery(path)
```
Comment (freddygv) [inline dependency/vault_write_test.go]: should this fail on err?

File: dependency/vault_write_test.go [hunk 2]
```diff
@@ -71,6 +71,69 @@ func TestNewVaultWriteQuery(t *testing.T) {
...
```
Comment (freddygv) [inline dependency/vault_write_test.go]: same as above

File: dependency/vault_write.go [hunk 3]
```diff
@@ -168,12 +170,20 @@ func (d *VaultWriteQuery) writeSecret(clients *ClientSet, opts *QueryOptions) (*
...
+	if isv2 {
+		data = map[string]interface{}{"data": d.data}
+	}
+
+	vaultSecret, err := clients.Vault().Logical().Write(d.path, data)
 	if err != nil {
 		return nil, errors.Wrap(err, d.String())
 	}
+	// always nil when KVv1 engine, when isv2==false
 	if vaultSecret == nil {
```
Comment (freddygv) [inline dependency/vault_write.go]: why not combine these with `if vaultSecret == nil && isv2`?

File: dependency/vault_write_test.go [hunk 4]
```diff
@@ -71,6 +71,69 @@ func TestNewVaultWriteQuery(t *testing.T) {
 	}
 }
 
+func TestVaultWriteSecretKV_Fetch(t *testing.T) {
+	t.Parallel()
+
+	// trigger a nil-pointer-deref panic in wq.Fetch() with KVv1 due to
```
Comment (freddygv) [inline dependency/vault_write_test.go]: these comments may seem cryptic to people looking at this test after the fix is in

File: dependency/vault_write_test.go (hunk 1 above)
Comment (eikenb) [reply dependency/vault_write_test.go]: Sure should. Seems like the linter should pick out unhandled err values like that. I overlooked it as I was focused on using it to test the value read, not so much the reader.

File: dependency/vault_write_test.go (hunk 4 above)
Comment (eikenb) [reply dependency/vault_write_test.go]: Is there some way I could make the clearer? There are times when I'm looking at a test and wondering why it was added... so I thought I'd start putting in small comments to see if that helped.

Comment (freddygv) [reply dependency/vault_write_test.go]: Yeah that would be useful. You could add a reference to the GH issue for context.

Something like:
```
// previously triggered a nil-pointer-deref panic in wq.Fetch() with KVv1 due to
// writeSecret() returning nil for vaultSecret
// see #1255
```

File: dependency/vault_write.go (hunk 3 above)
Comment (eikenb) [reply dependency/vault_write.go]: Because I missed that refactor. Thanks.

File: dependency/vault_write_test.go (hunk 4 above)
Comment (eikenb) [reply dependency/vault_write_test.go]: Thanks. I'll use GH-1252 to make it clear it is a github issue. 

I'll make a similar change (past tense + issue reference) to the other test's comment as well.

Thanks!

**Pull request merged**